File: Antlr/src/main/scala/io/github/chaosunity/casc/parsing/expression/ConditionalExpression.scala
package io.github.chaosunity.casc.parsing.expression

import io.github.chaosunity.casc.parsing.LogicalOp.LogicalOp
import io.github.chaosunity.casc.parsing.`type`.BuiltInType

class ConditionalExpression(val left: Expression,
                            val right: Expression,
                            val opcode: LogicalOp) extends Expression(BuiltInType.BOOLEAN, false) {
}

File: src/main/kotlin/io/github/chaosunity/casc/parsing/node/expression/ConditionalExpression.kt
package io.github.chaosunity.casc.parsing.node.expression

import io.github.chaosunity.casc.parsing.LogicalOp
import io.github.chaosunity.casc.parsing.type.BuiltInType
import io.github.chaosunity.casc.parsing.type.Type

data class ConditionalExpression(val leftExpression: Expression<*>, val rightExpression: Expression<*>, val logicalOp: LogicalOp) : Expression<ConditionalExpression> {
    override val type: Type = BuiltInType.BOOLEAN
    override val negative: Boolean = false
}
